kratos/solving_strategies/strategies/explicit_solving_strategy_runge_kutta_4.h

```diff
@@ -0,0 +1,414 @@
+        int RebuildLevel = 0)
+        : BaseType(rModelPart, pExplicitBuilder, MoveMeshFlag, RebuildLevel)
+    {
+    }
+
+    /**
+     * @brief Default constructor.
+     * @param rModelPart The model part to be computed
+     * @param MoveMeshFlag The flag to set if the mesh is moved or not
+     */
+    explicit ExplicitSolvingStrategyRungeKutta4(
+        ModelPart &rModelPart,
+        bool MoveMeshFlag = false,
+        int RebuildLevel = 0)
+        : BaseType(rModelPart, MoveMeshFlag, RebuildLevel)
+    {
+    }
+
+    /** Copy constructor.
+     */
+    ExplicitSolvingStrategyRungeKutta4(const ExplicitSolvingStrategyRungeKutta4 &Other) = delete;
+
+    /** Destructor.
+     */
+    virtual ~ExplicitSolvingStrategyRungeKutta4() = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "ExplicitSolvingStrategyRungeKutta4";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream &rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream &rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    void SolveWithLumpedMassMatrix() override
+    {
+        // Get the required data from the explicit builder and solver
+        const auto p_explicit_bs = BaseType::pGetExplicitBuilder();
+        auto& r_dof_set = p_explicit_bs->GetDofSet();
+        const unsigned int dof_size = p_explicit_bs->GetEquationSystemSize();
+        const auto& r_lumped_mass_vector = p_explicit_bs->GetLumpedMassMatrixVector();
+
+        // Set the auxiliary RK vectors
+        LocalSystemVectorType u_n(dof_size); // TODO: THIS IS INEFICCIENT. CREATE A UNORDERED_SET WITH THE IDOF AND VALUE AS ENTRIES. THIS HAS TO BE OPTIONAL
+        LocalSystemVectorType rk_k1(dof_size);
+        LocalSystemVectorType rk_k2(dof_size);
+        LocalSystemVectorType rk_k3(dof_size);
+        LocalSystemVectorType rk_k4(dof_size);
+
+        // Perform the RK 4 update
+        const double dt = BaseType::GetDeltaTime();
+        KRATOS_ERROR_IF(dt < 1.0e-12) << "ProcessInfo DELTA_TIME is close to zero." << std::endl;
+
+        // Set the previous step solution in the current buffer position
+        // Note that we set the 0 position of the buffer to be equal to the values in step n (not n+1)
+        // Additionally, we save in an auxiliary vector the value of the fixed DOFs, which is also taken from the previous time step
+        IndexPartition<int>(r_dof_set.size()).for_each(
```

Comment:
better use `BlockPartition`, index is not needed

Reply:
Done.

Reply:
Is required to access the `u_n` vector in case the DOF is fixed.